### PR TITLE
Use case-insensitive sorting in tables

### LINKED ISSUE #1527: Make sorting in tables case-insensitive

### PR DESCRIPTION
**Pull request by okahilak**
Closes #1527

# Definition of Done / Review checklist

## Reviewability
- [ ] link to issue
- [ ] note if PR is on top of other PR
- [ ] note if related change in rems-deploy repo
- [ ] consider adding screenshots for ease of review

## API
- [ ] API is documented and shows up in Swagger UI
- [ ] API is backwards compatible or completely new
- [ ] Events are backwards compatible

## Documentation
- [ ] update changelog if necessary
- [ ] add or update docstrings for namespaces and functions
- [ ] components are added to guide page
- [ ] documentation _at least_ for config options (i.e. docs folder)
- [ ] ADR for major architectural decisions or experiments

## Different installations
- [ ] new configuration options added to rems-deploy repository
- [ ] instance specific translations (i.e. LBR kielivara)

## Testing
- [ ] complex logic is unit tested
- [ ] valuable features are integration / browser / acceptance tested automatically

## Accessibility
- [ ] all icons have the aria-label attribute
- [ ] all fields have a label
- [ ] errors are linked to fields with aria-describedby
- [ ] contrast is checked
- [ ] conscious decision about where to move focus after an action

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [ ] no critical TODOs left to implement

### CODE REVIEW

File: src/cljs/rems/table.cljs [hunk 1]
```diff
@@ -152,8 +152,8 @@
    (->> rows
         (sort-by #(get-in % [(:sort-column sorting) :sort-value])
```
Comment (opqdonut) [inline src/cljs/rems/table.cljs]: [nit] you could put str/lower-case here in the keyfn, instead of below in the comparison

Comment (okahilak) [reply src/cljs/rems/table.cljs]: Done.

File: src/cljs/rems/table.cljs [hunk 2]
```diff
@@ -68,14 +67,18 @@
     m
     (assoc m k (make-default m))))
 
-(defn- apply-row-defaults [row]
+(defn apply-row-defaults [row]
```
Comment (okahilak) [inline src/cljs/rems/table.cljs]: I had to change this function to public to test it outside the namespace -- otherwise `lein doo once` gives the warning `WARNING: var: rems.table/apply-row-defaults is not public at line 12 test/cljs/rems/test_table.cljs`.

Are there any better solutions?

**Comment by okahilak** [comment]
The logic for lower-casing was in a slightly wrong place previously, now it fell inside a function that was already being tested. I moved the tests into their own testing namespace (maybe not necessary but I started doing it before realizing that I don't need another deftest but can use the existing ones) and added an explicit test for string and non-string values.

File: src/cljs/rems/table.cljs (hunk 2 above)
Comment (opqdonut) [reply src/cljs/rems/table.cljs]: The usual trick is to refer to privates like `@#'rems.table/apply-row-defaults`, but I'm not sure if that works in cljs.
Another option is to have the test in this file.
Making the function public is fine.

File: test/cljs/rems/test_table.cljs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                                :foo {:td [:td.foo [:button "Button"]]}}))))
+  (testing ":sort-value is normalized to lowercase"
+    (is (= {:key 123
+            :foo {:value "FooBar"
+                  :sort-value "foobar"
+                  :display-value "FooBar"
+                  :filter-value ""
+                  :td [:td ""]}}
+           (apply-row-defaults {:key 123
+                                :foo {:value "FooBar"
```
Comment (opqdonut) [inline test/cljs/rems/test_table.cljs]: But if you have `:sort-value "FooBar"`, it won't get lower-cased. Is that intentional?

Comment (okahilak) [reply test/cljs/rems/test_table.cljs]: Nope, I'll fix..

Comment (okahilak) [reply test/cljs/rems/test_table.cljs]: Hmm, even though that wasn't intentional, I think it could work that way. Currently, :filter-value works analogously, and it makes sense to use the same logic for both, i.e., if someone wants to filter (or sort) explicitly by a non-lower case string, it can be done, but the default is lower-cased.

I added test cases for that for both :filter-value and :sort-value, so that the behavior is more explicit and kind of documented in the test cases.

So, in short, I'd like to keep :filter-value and :sort-value analogous in this regard, and wouldn't want to change the existing behavior of :filter-value without strong enough reasons.

Comment (opqdonut) [reply test/cljs/rems/test_table.cljs]: that sounds reasonable
